pkg/minikube/update/update.go

```diff
@@ -49,64 +50,86 @@ const (
 	githubRepo  = "minishift"
 )
 
-var (
-	lastUpdateCheckFilePath = constants.MakeMiniPath("last_update_check")
-)
-
-func MaybeUpdateFromGithub(output io.Writer) {
+func MaybeUpdateFromGithub(output io.Writer) error {
 	localVersion, err := version.GetSemverVersion()
 	if err != nil {
 		glog.Errorln(err)
-		return
+		return err
 	}
-	MaybeUpdate(output, githubOwner, githubRepo, githubRepo, lastUpdateCheckFilePath, localVersion)
+
+	latestVersion, err := getLatestVersionFromGitHub(githubOwner, githubRepo)
+	if err != nil {
+		return err
+	}
+
+	if checkLatestVersion(localVersion, latestVersion) {
```

Comment:
_isNewerVersion_ or something like this. _checkLatestVersion_ does not really make clear what it does